libphx/luajit-ffi-gen/tests/basic_test.rs

```diff
@@ -0,0 +1,80 @@
+use luajit_ffi_gen::luajit_ffi;
+
+#[derive(Default, Clone)]
+pub struct Data {
+    pub val: bool,
+}
+
+#[derive(Default)]
+pub struct MyStruct {
+    val_u32: u32,
+    val_f32: f32,
+    // val_str: String,
+    val_data: Data,
+}
+
+// NOTE: remove 'no_lua_ffi' parameter to see generated Lua file. Do not commit it!!!
+#[luajit_ffi(name = "My_Struct", no_lua_ffi = true)]
+impl MyStruct {
+    pub fn func1(&self) {}
+    pub fn func2(&mut self) {}
+
+    #[bind(name = "FUNC3")]
+    pub fn func3() {}
+
+    pub fn set_u32(&mut self, val: u32) {
+        self.val_u32 = val;
+    }
+
+    pub fn get_u32(&self) -> u32 {
+        self.val_u32
+    }
+
+    pub fn set_f32(&mut self, val: &f32) {
+        self.val_f32 = *val;
+    }
+
+    pub fn get_f32(&self) -> f32 {
+        self.val_f32
+    }
+
+    // pub fn set_str(&mut self, val: &str) {
```

Comment:
Why are these commented out?

Reply:
There was a problem with exporting **static_string!** macro. It was circular dependency issue: phx depends on proc crate macro crate so we cannot add phx as dependency to it.
If I won't find some quick workaround I'll probably remove this part.

Reply:
Tried - it didn't work because of the issue I mentioned above.
I'll come to this after code restructuring